test/spmd/tensor/test_matrix_ops.py

```diff
@@ -127,6 +130,119 @@ def test_t(self):
         self.assertEqual(tranposed_mat2.size(), torch.Size([12, 8]))
         self.assertEqual(tranposed_mat2.placements, shard_spec)
 
+    # baddbmm introduces nan occasionally on CPU: https://github.com/pytorch/pytorch/issues/80588
+    @with_comms
+    @skip_if_lt_x_gpu(TEST_GPU_NUM)
+    def test_sharded_baddbmm(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        tensor = torch.rand(4, 4, 8, device=self.device_type)
+        batch_1 = torch.rand(4, 4, 8, device=self.device_type)
+        batch_2 = torch.rand(4, 8, 8, device=self.device_type)
+
+        def test_placement_comb(
+            tensor_placements: Sequence[Placement],
+            batch_1_placements: Sequence[Placement],
+            batch_2_placements: Sequence[Placement],
+            beta: int,
+            alpha: int,
+        ) -> None:
+            tensor_dt = distribute_tensor(
+                tensor, device_mesh, tensor_placements
+            )
+            batch_1_dt = distribute_tensor(
+                batch_1, device_mesh, batch_1_placements
+            )
+            batch_2_dt = distribute_tensor(
+                batch_2, device_mesh, batch_2_placements
+            )
+            new_dt = cast(
```

Comment:
Not needed in this PR, but we also need a backward test for baddbmm and bmm since both are used in ViT.